controller/gen/apis/policy/v1alpha1/types.go

```diff
@@ -0,0 +1,131 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +groupName=policy.linkerd.io
+
+type AuthorizationPolicy struct {
+	// TypeMeta is the metadata for the resource, like kind and apiversion
+	metav1.TypeMeta `json:",inline"`
+
+	// ObjectMeta contains the metadata for the particular object, including
+	// things like...
+	//  - name
+	//  - namespace
+	//  - self link
+	//  - labels
+	//  - ... etc ...
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the custom resource spec
+	Spec AuthorizationPolicySpec `json:"spec,"`
+}
+
+// AuthorizationPolicySpec specifies a, AuthorizationPolicySpec resource.
+type AuthorizationPolicySpec struct {
+	// TargetRef references a resource to which the authorization policy applies.
+	TargetRef gatewayapiv1alpha2.PolicyTargetReference `json:"targetRef,omitempty"`
+
+	// RequiredAuthenticationRefs enumerates a set of required authentications
+	RequiredAuthenticationRefs []gatewayapiv1alpha2.PolicyTargetReference `json:"required_authentication_refs,omitempty"`
```

Comment:
I've just stumbled upon this while working with the new API features available in 2.12.0. Is there a particular reason why this JSON field was named using the snake_case convention rather than the camelCase one? This goes against the [Kubernetes API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions). I would be more than happy to submit a patch unless there's solid reasoning behind it.